docs/listing.md

```diff
@@ -0,0 +1,15 @@
+# Listing
+
+## Listing Types
+
+We use three different types of metadata listing for various operations.
+Here are the scenarios we use each for:
+
+- 'Delimiter' - when no versions are involved, such as listing buckets
```

Comment:
when no versions are possible in the bucket since it is a bucket only used internally and not exposed to a user. Namely, (1) to list objects in the "user's bucket" to respond to a GET SERVICE request and (2) to do internal listings on an MPU shadow bucket to complete multipart upload operations.